src/sinks/mod.rs

```diff
@@ -66,8 +67,6 @@ pub mod loki;
 pub mod nats;
 #[cfg(feature = "sinks-new_relic")]
 pub mod new_relic;
-#[cfg(feature = "sinks-new_relic_logs")]
-pub mod new_relic_logs;
```

Comment:
Hmm, not sure what you mean?

Reply:
Ah I have https://github.com/vectordotdev/vector/issues/14210 on my todo, if we're deprecating this :soon: perhaps I should skip it. Probably a better question for @jszwedko

Reply:
We haven't officially deprecated this sink yet. We could do so in 0.25.0 and plan to remove in 0.26.0.

Reply:
Well... that's only true if the config and behavior between the two is the same, it wasn't a direct alias. `new_relic` was a new sink that should have covered the same use-case, but I don't know if we need to publish a transition guide.

Reply:
Ah yeah, that's a really good point. I'll add this to the list of things to investigate.

Reply:
Ah, yes, that one wasn't an alias. It probably needs a short migration blurb. I'm ok with backporting that blurb to 0.24.0 since the deprecation was, as you noted, announced there.

Additionally we should mark it as deprecated in the docs: https://github.com/vectordotdev/vector/blob/05fc485f52eba72eb4bb68ab6d392753138f0f2d/website/cue/reference/components/sinks/new_relic.cue#L9

If you want to go the whole 9 yards, it should also output a warning if detected in user configuration per https://github.com/vectordotdev/vector/blob/master/docs/DEPRECATION.md#deprecation

Reply:
noting linked line is the _not_ deprecated sink.

Reply:
I added an upgrade guide entry for 0.25.0 explaining how to switch to `new_relic` which boils down to changing the component type and having to add a few additional settings: API type set to logs, specify an account ID, use a license key only instead of insert key vs license key.

There's not really a great way -- without a lot of extra work -- to actually emit a deprecation notice for a renamed sink. It's possible to do, but it's not quite the same as components with deprecated fields that can check if they've been set, and the new one hasn't been, etc.

Reply:
Could the sink itself just emit a warning if it is being constructed?

Reply:
(We could backport that warning into 0.24.1)

Reply:
Sure, that would also be doable.